sympy/matrices/decompositions.py

```diff
@@ -1122,6 +1121,222 @@ def _LUdecompositionFF(M):
+
+    >>> V * V.H
+    Matrix([
+    [1, 0],
+    [0, 1]])
+    >>> V.H * V
+    Matrix([
+    [1, 0],
+    [0, 1]])
+    >>> A == U * S * V.H
+    True
+
+    >>> C = Matrix([
+    ...         [1, 0, 0, 0, 2],
+    ...         [0, 0, 3, 0, 0],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 2, 0, 0, 0],
+    ...     ])
+    >>> U, S, V = C.SVD()
+
+    >>> V.H * V
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+    >>> V * V.H
+    Matrix([
+    [1/5, 0, 0, 0, 2/5],
+    [  0, 1, 0, 0,   0],
+    [  0, 0, 1, 0,   0],
+    [  0, 0, 0, 0,   0],
+    [2/5, 0, 0, 0, 4/5]])
+
+    If you want to augment the results to be a full orthogonal
+    decomposition, you should augment $V$ with an another orthogonal
+    column.
+
+    You are able to append an arbitrary standard basis that are linearly
+    independent to every other columns and you can run the Gram-Schmidt
+    process to make them augmented as orthogonal basis.
+
+    >>> V_aug = V.row_join(Matrix([[0,0,0,0,1],
+    ... [0,0,0,1,0]]).H)
+    >>> V_aug = V_aug.QRdecomposition()[0]
+    >>> V_aug
+    Matrix([
+    [0,   sqrt(5)/5, 0, -2*sqrt(5)/5, 0],
+    [1,           0, 0,            0, 0],
+    [0,           0, 1,            0, 0],
+    [0,           0, 0,            0, 1],
+    [0, 2*sqrt(5)/5, 0,    sqrt(5)/5, 0]])
+    >>> V_aug.H * V_aug
+    Matrix([
+    [1, 0, 0, 0, 0],
+    [0, 1, 0, 0, 0],
+    [0, 0, 1, 0, 0],
+    [0, 0, 0, 1, 0],
+    [0, 0, 0, 0, 1]])
+    >>> V_aug * V_aug.H
+    Matrix([
+    [1, 0, 0, 0, 0],
+    [0, 1, 0, 0, 0],
+    [0, 0, 1, 0, 0],
+    [0, 0, 0, 1, 0],
+    [0, 0, 0, 0, 1]])
+
+    Similarly we augment U
+
+    >>> U_aug = U.row_join(Matrix([0,0,1,0]))
+    >>> U_aug = U_aug.QRdecomposition()[0]
+    >>> U_aug
+    Matrix([
+    [0, 1, 0, 0],
+    [0, 0, 1, 0],
+    [0, 0, 0, 1],
+    [1, 0, 0, 0]])
+
+    >>> U_aug.H * U_aug
+    Matrix([
+    [1, 0, 0, 0],
+    [0, 1, 0, 0],
+    [0, 0, 1, 0],
+    [0, 0, 0, 1]])
+    >>> U_aug * U_aug.H
+    Matrix([
+    [1, 0, 0, 0],
+    [0, 1, 0, 0],
+    [0, 0, 1, 0],
+    [0, 0, 0, 1]])
+
+    We add 2 zero columns and one row to S
+
+    >>> S_aug = S.col_join(Matrix([[0,0,0]]))
+    >>> S_aug = S_aug.row_join(Matrix([[0,0,0,0],
+    ... [0,0,0,0]]).H)
+    >>> S_aug
+    Matrix([
+    [2,       0, 0, 0, 0],
+    [0, sqrt(5), 0, 0, 0],
+    [0,       0, 3, 0, 0],
+    [0,       0, 0, 0, 0]])
+
+
+
+    >>> U_aug * S_aug * V_aug.H == C
+    True
+
+    """
+
+    AH = A.H
+    m, n = A.shape
+    if m >= n:
+        V, S = (AH * A).diagonalize()
+
+        ranked = []
+        for i, x in enumerate(S.diagonal()):
+            if not x.is_zero:
+                ranked.append(i)
+
+        V = V[:, ranked]
+
+        Singular_vals = [sqrt(S[i, i]) for i in range(S.rows) if i in ranked]
+
+        S = S.zeros(len(Singular_vals))
+
+        for i in range(len(Singular_vals)):
+            S[i, i] = Singular_vals[i]
+
+        V, _ = V.QRdecomposition()
+        U = A * V * S.inv()
```

Comment:
No spaces around * and **.

Reply:
No, I don't think that PEP8 recommendation states anything about this
https://www.python.org/dev/peps/pep-0008/#other-recommendations